vespalib/src/vespa/fastos/linux_file.cpp

```diff
@@ -239,8 +239,8 @@ FastOS_Linux_File::internalWrite2(const void *buffer, size_t length)
         }
         if (writeRes > 0) {
             _filePointer += writeRes;
-            if (_filePointer > _cachedSize) {
-                _cachedSize = _filePointer;
+            if (_filePointer > _cachedSize.load(std::memory_order_relaxed)) {
+                _cachedSize.store(_filePointer, std::memory_order_release);
```

Comment:
If this can happen from multiple threads more or less at the same time, should this be a (relaxed) `compare_exchange` loop where we retry until the cached size is at least as high as `_filePointer`? This would be quite a bit more expensive, so maybe it's not required to be that up-to-date?

Reply:
Are there any transitive memory visibility dependencies that require this to be `release` rather than `relaxed`?